tests/migrations/test_operations.py

```diff
@@ -1094,6 +1094,41 @@ def test_add_binaryfield(self):
         self.assertEqual(bytes(pony.digits), b"42")
         self.assertEqual(bytes(pony.quotes), b'"\'"')
 
+    def test_add_foreignkey(self):
+        """
+        Tests that adding a ForeignKey creates the corresponding constraint.
+
+        Also tests that, for backends that support it, creating a FK field with
+        a constraint only takes a single statement without requiring a deferred
+        ALTER TABLE.
+        """
+        project_state = self.set_up_test_model('test_adfkfl', second_model=True)
+
+        migration = Migration('add_pony_stable_fk', 'test_adfkfl')
+        migration.operations = [
+            migrations.AddField(
+                'Pony',
+                'stable',
+                models.ForeignKey('test_adfkfl.Stable', on_delete=models.CASCADE),
+            )
+        ]
+        with connection.schema_editor() as editor:
+            migration.apply(project_state, editor)
+
+        self.assertFKExists('test_adfkfl_pony', ['stable_id'], ('test_adfkfl_stable', 'id'))
+
+        if connection.vendor in ('mysql', 'oracle', 'postgresql'):
```

Comment:
This should use a database feature `django/db/backends/base/features.py` so that other third-party database backends can also use the test.

Reply:
I wondered if someone was going to say that!

Any opinion what the feature should be named? Maybe `can_create_fk_constraints_inline`?

Reply:
`can_create_inline_fk` would mirror the name of `sql_create_inline_fk`.